### PR TITLE
Fix/zephyr includes

### PR DESCRIPTION
**Pull request by uLipe**
This filters the zephyr versions using kernel macros to decide when to namespace zephyr headers.

### CODE REVIEW

**Comment by uLipe** [comment]
@pablogs9 @Acuadros95 PTAL

**Comment by uLipe** [comment]
https://github.com/micro-ROS/micro_ros_zephyr_module/pull/82

**Comment by pablogs9** [comment]
Hello @uLipe I'm gonna close https://github.com/micro-ROS/micro_ros_zephyr_module/pull/74 in favor of this but I need some minor changes

File: modules/libmicroros/CMakeLists.txt [hunk 1]
```diff
@@ -48,6 +48,7 @@ externalproject_add(libmicroros_project
             COMPONENT_PATH=${MICROROS_DIR}
             ZEPHYR_BASE=${ZEPHYR_BASE}
             PROJECT_BINARY_DIR=${PROJECT_BINARY_DIR}
+            KERNEL_VERSION_MAJOR=${KERNEL_VERSION_MAJOR}
```
Comment (pablogs9) [inline modules/libmicroros/CMakeLists.txt:51]:
```suggestion
```

Not required

File: modules/libmicroros/CMakeLists.txt [hunk 2]
```diff
@@ -94,6 +95,8 @@ zephyr_library_sources(
    ${MICROROS_TRANSPORT_DIR}/microros_transports.c
 )
 
+zephyr_compile_definitions(KERNEL_VERSION_MAJOR=${KERNEL_VERSION_MAJOR})
+
```
Comment (pablogs9) [inline modules/libmicroros/CMakeLists.txt:99]:
```suggestion
```
Not required

File: modules/libmicroros/libmicroros.mk [hunk 3]
```diff
@@ -55,7 +55,8 @@ configure_toolchain: $(COMPONENT_PATH)/zephyr_toolchain.cmake.in
 		sed "s/@CMAKE_CXX_COMPILER@/$(subst /,\/,$(X_CXX))/g" | \
 		sed "s/@CMAKE_SYSROOT@/$(subst /,\/,$(COMPONENT_PATH))/g" | \
 		sed "s/@CFLAGS@/$(subst /,\/,$(CFLAGS_INTERNAL))/g" | \
-		sed "s/@CXXFLAGS@/$(subst /,\/,$(CXXFLAGS_INTERNAL))/g" \
+		sed "s/@CXXFLAGS@/$(subst /,\/,$(CXXFLAGS_INTERNAL))/g" | \
+		sed "s/@KERNEL_VERSION_MAJOR@/$(subst /,\/,$(KERNEL_VERSION_MAJOR))/g" \
```
Comment (pablogs9) [inline modules/libmicroros/libmicroros.mk:59]:
```suggestion
                sed "s/@CXXFLAGS@/$(subst /,\/,$(CXXFLAGS_INTERNAL))/g" \
```

File: modules/libmicroros/zephyr_toolchain.cmake.in [hunk 4]
```diff
@@ -26,7 +26,7 @@ SET(CMAKE_CXX_COMPILER_WORKS 1 CACHE INTERNAL "")
 set(CMAKE_C_COMPILER @CMAKE_C_COMPILER@)
 set(CMAKE_CXX_COMPILER @CMAKE_CXX_COMPILER@)
 
-set(CMAKE_C_FLAGS_INIT "@CFLAGS@ -DCLOCK_MONOTONIC=0" CACHE STRING "" FORCE)
-set(CMAKE_CXX_FLAGS_INIT "@CXXFLAGS@ -DCLOCK_MONOTONIC=0" CACHE STRING "" FORCE)
+set(CMAKE_C_FLAGS_INIT "@CFLAGS@ -DCLOCK_MONOTONIC=0 -DKERNEL_VERSION_MAJOR=@KERNEL_VERSION_MAJOR@" CACHE STRING "" FORCE)
+set(CMAKE_CXX_FLAGS_INIT "@CXXFLAGS@ -DCLOCK_MONOTONIC=0 -DKERNEL_VERSION_MAJOR=@KERNEL_VERSION_MAJOR@" CACHE STRING "" FORCE)
```
Comment (pablogs9) [inline modules/libmicroros/zephyr_toolchain.cmake.in:30]:
```suggestion
set(CMAKE_C_FLAGS_INIT "@CFLAGS@ -DCLOCK_MONOTONIC=0" CACHE STRING "" FORCE)
set(CMAKE_CXX_FLAGS_INIT "@CXXFLAGS@ -DCLOCK_MONOTONIC=0" CACHE STRING "" FORCE)
```

**Review by pablogs9 — COMMENTED** [review]
Minor changes to use only the way of version checking proposed by @uLipe

**Comment by uLipe** [comment]
@pablogs9 PTAL

**Review by pablogs9 — APPROVED** [review]
LGTM

**Comment by pablogs9** [comment]
@mergify backport foxy galactic main

**Pull request merged**